scraping/extract-magic-objects.py

```diff
@@ -122,7 +122,28 @@
             if "coût" in data:
                 element["Coût"] = data["coût"]
             
-            liste.append(element)
+            # si l'objet à plusieurs prix, alors créer les variations de l'objet
+            variations = re.split('(?<=\)),\s+|(?<=\))\s+ou\s+', element["Prix"]) 
+            nomBackup = element["Nom"]
+            if len(variations) > 1:
+                for variation in variations:
+                    try:
+                        price = re.search('^(.+? po) \(', variation).group(1)
+                    except:
+                        print("ERROR: Could not extract the variation detail for " + element["Nom"] + " '" + variation + "'")
+                        exit(1)
+                    try:
+                        detail = re.search('po \((.+?)\)$', variation).group(1)
+                    except:
+                        print("ERROR: Could not extract the variation detail for " + element["Nom"] + "'" + variation + "'")
+                        exit(1)
+                    element["Prix"] = price
+                    element["Nom"] = element["Nom"] + " (" + detail + ")"
+                    print("Creation de la variation " + element["Nom"])
+                    liste.append(element)
+                    element["Nom"] = nomBackup
```

Comment:
J'ai du ajuster cette partie pour faire une copie de l'élément plutôt que de le réutiliser à chaque itération car sinon tous les éléments se retrouvent avec le dernier nom changé.